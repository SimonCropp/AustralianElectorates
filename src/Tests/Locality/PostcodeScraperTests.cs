using System.IO;
using System.Linq;
using System.Threading.Tasks;
using VerifyXunit;
using Xunit;
using Xunit.Abstractions;

public class PostcodeScraperTests
{
    [Fact]
    [Trait("Category", "Integration")]
    public async Task Run()
    {
        var data = await PostcodeScraper.Run();
        File.Delete(DataLocations.LocalitiesPath);
        JsonSerializerService.Serialize(data, DataLocations.LocalitiesPath);
        await Verifier.Verify(data.Take(10));
    }

    [Fact]
    public Task Specific()
    {
        var place = AustraliaData.PostCodes.Single(x=>x.Key == "2612");
        return Verifier.Verify(PostcodeScraper.GetAECDataForPostcode(place));
    }

    public PostcodeScraperTests(ITestOutputHelper output) :
        base(output)
    {
    }
}