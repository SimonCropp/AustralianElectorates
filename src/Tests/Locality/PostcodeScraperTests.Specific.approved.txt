[
  {
    State: 'ACT',
    LocalityOrSuburb: 'BRADDON',
    Postcode: 2612,
    Electorate: 'CANBERRA'
  },
  {
    State: 'ACT',
    LocalityOrSuburb: 'CAMPBELL',
    Postcode: 2612,
    Electorate: 'CANBERRA'
  },
  {
    State: 'ACT',
    LocalityOrSuburb: 'REID',
    Postcode: 2612,
    Electorate: 'CANBERRA'
  },
  {
    State: 'ACT',
    LocalityOrSuburb: 'TURNER',
    Postcode: 2612,
    Electorate: 'CANBERRA'
  }
]